### ISSUE TITLE
Code Table Request - New Collection Object Attribute = life stage

### ISSUE THREAD

**Issue by Jegelewicz** [ArctosDB/arctos#4523]
**Goal**
From https://github.com/ArctosDB/arctos/issues/2881#issuecomment-1092104964

**Context**
From https://github.com/ArctosDB/arctos/issues/2881#issuecomment-1092104964

**Table**
https://arctos.database.museum/info/ctDocumentation.cfm?table=ctattribute_type

**Proposed Value**
life stage

**Proposed Definition**
~~The developmental stage of the life cycle of the organism represented.~~

One of the series of defined stages of the life cycle of the organism represented.

**Collection type**
Amph, Arc, Bird, Ento, ES, Fish, Herb, Herp, Host, Inv, Mamm, Para, Rept, Teach

**Attribute data type**
categorical

**Attribute value**
need new table = life_stage

**Available for Public View**
Yes

**Priority**
[  Please choose a priority-label to the right.  ]

### Available for Public View
_Most data are by default publicly available. Describe any necessary access restrictions._

yes

### Helpful Actions

- [x] Add the issue to the [Code Table Management Project](https://github.com/ArctosDB/arctos/projects/13#card-31628184).

- [x] Please reach out to anyone who might be affected by this change. Leave a comment or add this to the Committee agenda if you believe more focused conversation is necessary.

@ArctosDB/arctos-code-table-administrators 

## Approval
_All of the following must be checked before this may proceed._

_The [How-To Document](https://handbook.arctosdb.org/how_to/How-To-Manage-Code-Table-Requests.html) should be followed. Pay particular attention to terminology (with emphasis on consistency) and documentation (with emphasis on functionality). **No person should act in multiple roles**; the submitter cannot also serve as a Code Table Administrator, for example._

- [ ] Code Table Administrator[1] - check and initial, comment, or thumbs-up to indicate that the request complies with the how-to documentation and has your approval
- [ ] Code Table Administrator[2] - check and initial, comment, or thumbs-up to indicate that the request complies with the how-to documentation and has your approval
- [ ] DBA - The request is functionally acceptable. The term is not a functional duplicate, and is compatible with existing data and code.
- [ ] DBA - Appropriate code or handlers are in place as necessary. (ID_References, Media Relationships, Encumbrances, etc. require particular attention)

## Rejection
_If you believe this request should not proceed, explain why here. Suggest any changes that would make the change acceptable, alternate (usually existing) paths to the same goals, etc._

1. _Can a suitable solution be found here? If not, proceed to (2)_
2. _Can a suitable solution be found by Code Table Committee discussion? If not, proceed to (3)_
3. _Take the discussion to a monthly Arctos Working Group meeting for final resolution._

## Implementation
_Once all of the Approval Checklist is appropriately checked and there are no Rejection comments, or in special circumstances by decree of the Arctos Working Group, the change may be made._

- [ ] Review everything one last time.  Ensure the How-To has been followed. Ensure all checks have been made by appropriate personnel.

- [ ] Add or revise the code table term/definition as described above. Ensure the URL of this Issue is included in the definition. URLs should be included *_as text_*, separated by spaced pipes. Do not include HTML in definitions.

_Close this Issue._

_**DO NOT** modify Arctos Authorities in any way before all points in this Issue have been fully addressed; data loss may result._

## Special Exemptions

_In very specific cases and by prior approval of The Committee, the approval process may be skipped, and implementation requirements may be slightly altered. Please note here if you are proceeding under one of these use cases._

1. _Adding an existing term to additional collection types may proceed immediately and without discussion, but doing so may also subject users to future cleanup efforts. If time allows, please review the term and definition as part of this step._
2. _The Committee may grant special access on particular tables to particular users. This should be exercised with great caution only after several smooth test cases, and generally limited to "taxonomy-like" data such as International Commission on Stratigraphy terminology._

**Comment by Jegelewicz** [comment]
Code table says go

**Comment by Jegelewicz** [comment]
Create a Google sheet to clean up definitions for controlled values

**Comment by Jegelewicz** [comment]
Above is already in place - see https://github.com/ArctosDB/arctos/issues/2881#issuecomment-1092107887

Closing

**Comment by dustymc** [comment]
eep/wut...

This is going to greatly complicate https://github.com/ArctosDB/arctos-dev/issues/24


https://arctos.database.museum/info/ctDocumentation.cfm?table=ctattribute_code_tables says this is using https://arctos.database.museum/info/ctDocumentation.cfm?table=ctage_class (wut-wut?!) so I recommend

1. Flip all https://arctos.database.museum/info/ctDocumentation.cfm?table=ctattribute_type#life_stage to https://arctos.database.museum/info/ctDocumentation.cfm?table=ctattribute_type#age_class
2. Drop the attribute
3. https://github.com/ArctosDB/arctos-dev/issues/24
4. Move things back (slowly and under robust definitions, as discussed by [code tables committee](https://docs.google.com/document/d/1eZ_S8VTp-m5Wrbb6cAK96ODPZpt2-Hvnvt6yfU9qVwg/edit#heading=h.3jdyx487bg21))


I don't think any of that changes any data so I'll proceed in the next ~week (whenever this becomes a problem) unless there's some functional objection.

Data:

[temp_ls.csv.zip](https://github.com/ArctosDB/arctos/files/13836287/temp_ls.csv.zip)

Summary:

```

  guid_prefix  | count 
---------------+-------
 UMNH:Mamm     |     2
 MSB:Herp      |    38
 CHAS:Teach    |     7
 MSB:Para      |  2181
 MSB:Host      |    31
 DMNS:Inv      |     2
 UAM:Ento      |     8
 CHAS:Ento     |     1
 ASUMZ:Bivalve |    14
 UMZM:Bird     |     1
```

Users:



 @campmlc
 @DerekSikes
 @jtgiermakowski
 @wellerjes
 @acdoll
 @genevieve-anderegg
 @msbparasites
 @kderieg322079
 @sharpphyl
 @adhornsby
 @droberts49
(11 rows)

**Comment by Jegelewicz** [comment]
eep indeed - that's what I said when I looked at the attribute - sorry this is such a pain....

**Comment by dustymc** [comment]
@Jegelewicz can you beef up this definition (or start a new issue, maybe that's cleaner, IDK) - also an issue for anything that will be immediately migrated, if there's anything.

**Comment by Jegelewicz** [comment]
> can you beef up this definition

What needs beefing up? This definition is backwarsd engineered from https://en.wikipedia.org/wiki/Biological_life_cycle because the only definitions to be found for life stage are human centric (or product focused...). I find the proposed definition sufficient. @ArctosDB/arctos-code-table-administrators please advise.

**Comment by Jegelewicz** [comment]
I have updated the request to include the approval check boxes.

**Comment by dustymc** [comment]
IDK - vague memories that we were looking for "better-defined than age class" and maybe we're there....

https://docs.google.com/document/d/1eZ_S8VTp-m5Wrbb6cAK96ODPZpt2-Hvnvt6yfU9qVwg/edit#heading=h.3jdyx487bg21

@ccicero @ebraker @Nicole-Ridgwell-NMMNHS

**Comment by dustymc** [comment]
See also https://github.com/ArctosDB/arctos/issues/7298

**Comment by dustymc** [comment]
Re https://github.com/ArctosDB/arctos/issues/4523#issuecomment-1877896956 - of course nothing can be that simple, there are duplicates. This is them, should be nothing lost in them being purged.

[temp_dup_lifestage.csv.zip](https://github.com/ArctosDB/arctos/files/13971150/temp_dup_lifestage.csv.zip)

--------------

> approval check boxes

I don't think I can let that stop an update.
